lib/silencer/rails/logger.rb

```diff
@@ -35,13 +35,13 @@ def initialize(app, *args)
       end
 
       def call(env)
-        old_logger_level     = ::Rails.logger.level
-        ::Rails.logger.level = ::Logger::ERROR if silence_request?(env)
-
-        super
-      ensure
-        # Return back to previous logging level
-        ::Rails.logger.level = old_logger_level
+        if silence_request?(env)
+          ::Rails.logger.silence do
```

Comment:
Maybe we might need to check if `Rails.logger.respond_to?(:silence)` here and falling back to adjusting the log level when it does not.